source/geod24/LocalRest.d

```diff
@@ -772,8 +818,398 @@ public final class RemoteAPI (API) : API
+            else static if (isImplicitlyConvertible!(param2[i], param1[i]))
+                enum isParamsImplicitlyConvertible = isParamsImplicitlyConvertible!(F1,
+                        F2, i + 1);
+            else
+                enum isParamsImplicitlyConvertible = false;
+        }
+
+        enum isSpawnable = isCallable!F && is(ReturnType!F == void)
+                && isParamsImplicitlyConvertible!(F, void function(LocalTransceiver, T))
+                && (isFunctionPointer!F || !hasUnsharedAliasing!F);
+    }
+
+    private LocalTransceiver spawnThread(F, T...)(F fn, T args)
+    if (isSpawnable!(F, T))
+    {
+        static assert(!hasLocalAliasing!(T), "Aliases to mutable thread-local data not allowed.");
+
+        auto thread_scheduler = new ThreadScheduler();
+        auto transceiver = new LocalTransceiver();
+        thread_scheduler.spawn({
+            thisLocalTransceiver = transceiver;
+            fn(transceiver, args);
+        });
+
+        return transceiver;
+    }
+}
+
+
+/*******************************************************************************
+
+    Transceiver device required for message exchange between threads.
+    Send and receive data requests, responses, commands, etc.
+
+*******************************************************************************/
+
+private class LocalTransceiver : Transceiver
+{
+
+    /// Channel of Request
+    public Channel!Message chan;
+
+    /// Ctor
+    public this () @safe nothrow
+    {
+        this.chan = new Channel!Message(64*1024);
+    }
+
+
+    /***************************************************************************
+
+        It is a function that accepts Message
+
+        Params:
+            msg = The `Message` to send.
+
+    ***************************************************************************/
+
+    public void send (T) (T msg) @trusted
+    in
+    {
+        assert(is (T == Message), "Only Message can be used.");
+    }
+    do
+    {
+        this.chan.send(msg);
+    }
+
+
+    /***************************************************************************
+
+        Return the received message.
+
+        Returns:
+            A received `Message`
+
+    ***************************************************************************/
+
+    public bool receive (T) (T *msg) @trusted
+    in
+    {
+        assert(is (T == Message), "Only Message can be used.");
+    }
+    do
+    {
+        return this.chan.receive(msg);
+    }
+
+
+    /***************************************************************************
+
+        Return the received message.
+
+        Params:
+            msg = The `Message` pointer to receive.
+
+        Returns:
+            Returns true when message has been received. Otherwise false
+
+    ***************************************************************************/
+
+    public bool tryReceive (T) (T *msg) @trusted
+    in
+    {
+        assert(is (T == Message), "Only Message can be used.");
```

Comment:
If `T` always needs to be `Message`, it means the function should not be templated